tests/stress/helpers.go

```diff
@@ -65,74 +65,60 @@ func compareChainHeadsWithRetry(t *testing.T, nodes []*utils.Node) error {
 
 // compareBlocksByNumber calls getBlockByNumber for each node in the array
 // it returns a map of block hashes to node key names, and an error if the hashes don't all match
-func compareBlocksByNumber(t *testing.T, nodes []*utils.Node, num string) (map[common.Hash][]string, error) {
-	hashes := make(map[common.Hash][]string)
-	var errs []error
-	var mapMu sync.Mutex
-	var wg sync.WaitGroup
-	wg.Add(len(nodes))
+func compareBlocksByNumber(ctx context.Context, t *testing.T, nodes []*utils.Node,
+	num string) (hashToKeys map[common.Hash][]string) {
+	type resultContainer struct {
```

Comment:
nit:
```suggestion
	type result struct {
```

Reply:
If I rename to `result` `govet` complains about shadowing with the `result`-named variable below.
And I know I could rename that variable below from `result` to `r`, but that defeats my crusade against single letter variable names :crossed_swords: (except `i` for indexes). If you have a better name, feel free to say it :smile: 

Also that's an inlined type so it won't pollute the API so it's not a big deal either.